cdap-common/src/main/java/co/cask/cdap/common/service/AbstractRetryableScheduledService.java

```diff
@@ -133,6 +133,16 @@ protected final void shutDown() throws Exception {
     }
   }
 
+  /**
+   * Called when the RetryStrategy used by this class returns a negative number, indicating to abort the operation.
```

Comment:
This is not clear what's the purpose of this method is. This method is give the chance for sub-class to reset the retry count or to terminate the service (by throwing exception).